docs/01_nodeos/02_usage/03_development-environment/10_local-single-node-testnet-consensus.md

```diff
@@ -0,0 +1,202 @@
+...
+```
+
+At this point, `nodeos` is running with a single producer, `eosio`.
+
+### 3. Preactivate Protocol Features
+
+All of the protocol upgrade features introduced in v1.8 and on subsequent versions require a special protocol feature, also known as `PREACTIVATE_FEATURE`, to be activated.
+
+To activate the special protocol `PREACTIVATE_FEATURE` run the following command  from a terminal:
+
+```sh
+curl --request POST \
+    --url http://127.0.0.1:8888/v1/producer/schedule_protocol_feature_activations \
+    -d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}'
+```
+
+### 4. Get The System Smart Contracts
+
+All of the protocol upgrade features introduced in v1.8 and on subsequent versions also require an updated version of the system smart contract which can make use of those protocol features.
+
+Two updated reference system smart contracts, `eosio.boot` and `eosio.bios`, are available in both source and binary form within the [`eos`](https://github.com/EOSIO/eos.git) repository. You can build them from source or deploy the binaries directly.
+
+#### 4.1 Use The Prebuilt System Smart Contracts
+
+To use the prebuilt system smart contract execute the following commands from a terminal:
+
+```sh
+cd ~
+git clone https://github.com/EOSIO/eos.git
+cd ./eos/contracts/contracts/contracts/
+pwd
+```
+
+Note the path printed at the command prompt, we will refer to it later as `EOSIO_SYSTEM_CONTRACTS_DIRECTORY`.
+
+Alternatively you can build the system smart contracts from source with the following commands:
+
+```sh
+cd ~
+git clone https://github.com/EOSIO/eos.git
+cd ./eos/contracts/contracts/
+mkdir build
+cd build
+cmake ..
+make
+pwd
+```
+
+### 5. Install eosio.boot System Contract
+
+To install the `eosio.boot` system contract execute the following command from a terminal. Make sure you replace the `EOSIO_SYSTEM_CONTRACTS_DIRECTORY` with the directory path where the `eosio.boot.wasm` and `eosio.boot.abi` files are located.
+
+```sh
+cleos set contract eosio EOSIO_SYSTEM_CONTRACTS_DIRECTORY/eosio.boot/bin/ eosio.boot.wasm eosio.boot.abi
+```
+
+You should see something similar to the following output:
+
+```sh
+Reading WAST/WASM from /users/documents/eos/contracts/contracts/eosio.boot/build/eosio.boot.wasm...
+Using already assembled WASM...
+Publishing contract...
+executed transaction: 2150ed87e4564cd3fe98ccdea841dc9ff67351f9315b6384084e8572a35887cc  39968 bytes  4395 us
+#         eosio <= eosio::setcode               {"account":"eosio","vmtype":0,"vmversion":0,"code":"0061736d0100000001be023060027f7e0060067f7e7e7f7f...
+#         eosio <= eosio::setabi                {"account":"eosio","abi":{"types":[],"structs":[{"name":"buyrambytes","base":"","fields":[{"name":"p...
+```
+
+### 6. Activate The Remaining Protocol Features
```

Comment:
Minor:
"The" need not be capitalized as per Title-style Capitalization rules if "The" occurs in the middle of a title.